internal/resources/cloud/data_source_private_data_source_connect_network.go

```diff
@@ -0,0 +1,130 @@
+package cloud
+
+import (
+	"context"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var dataSourcePrivateDataSourceConnectNetworksName = "grafana_cloud_private_data_source_connect_networks"
+
+func datasourcePrivateDataSourceConnectNetworks() *common.DataSource {
+	return common.NewDataSource(
+		common.CategoryCloud,
+		dataSourcePrivateDataSourceConnectNetworksName,
+		&PrivateDatasourceConnectNetworksDataSource{},
+	)
+}
+
+type PrivateDatasourceConnectNetworksDataSource struct {
```

Comment:
Can we be consistent about typing on `Datasource` please?

Reply:
I think it would be OK to use PDC in the type names too - up to you.

Reply:
Switched to using PDC